source/common/rate_limiter_impl.cc

```diff
@@ -166,15 +163,32 @@ GraduallyOpeningRateLimiterFilter::GraduallyOpeningRateLimiterFilter(
     : FilteringRateLimiterImpl(
           std::move(rate_limiter),
           [this]() {
-            if (elapsed() < ramp_time_) {
-              const double chance_percentage =
-                  100.0 - (static_cast<double>(ramp_time_.count() - elapsed().count()) /
-                           (ramp_time_.count() * 1.0)) *
-                              100.0;
-              return std::round(provider_->getValue() / 10000.0) <= chance_percentage;
+            const auto elapsed_time = elapsed();
+            if (elapsed_time < ramp_time_) {
+              // We want to linearly increase the probability of returning true
+              // below. We can derive that from the elapsed fraction of ramp_time.
+              const double probability =
+                  1.0 - static_cast<double>(ramp_time_.count() - elapsed_time.count()) /
+                            (ramp_time_.count() * 1.0);
+              // Get a random number r, where 0 < r ≤ 1.
+              const double random_between_0_and_1 = 1.0 * provider_->getValue() / provider_->max();
```

Comment:
Thank you, this is easier to follow with the new comments and the min() and max() methods.